coderd/httpapi/httperror/responserror.go

```diff
@@ -17,3 +21,48 @@ func IsResponder(err error) (Responder, bool) {
 	}
 	return nil, false
 }
+
+func NewResponseError(status int, resp codersdk.Response) error {
+	return &responseError{
+		status:   status,
+		response: resp,
+	}
+}
+
+func WriteResponseError(ctx context.Context, rw http.ResponseWriter, err error) {
+	if responseErr, ok := IsResponder(err); ok {
+		code, resp := responseErr.Response()
+
+		httpapi.Write(ctx, rw, code, resp)
+		return
+	}
+
+	httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+		Message: "Internal server error",
+		Detail:  err.Error(),
+	})
+}
+
+type responseError struct {
+	status   int
```

Comment:
How about embedding or extending `codersdk.Error` instead? It already has these fields.